manifests/qpid.pp

```diff
@@ -1,34 +1,47 @@
 # Katello qpid Config
 class katello::qpid (
-  $client_cert,
-  $client_key,
-  $katello_user            = $::katello::user,
-  $candlepin_event_queue   = $::katello::candlepin_event_queue,
-  $candlepin_qpid_exchange = $::katello::candlepin_qpid_exchange,
+  String $katello_user = $::katello::user,
+  String $candlepin_event_queue = $::katello::candlepin_event_queue,
+  String $candlepin_qpid_exchange = $::katello::candlepin_qpid_exchange,
+  Optional[Integer[0, 5000]] $session_unacked = $::katello::qpid_session_unacked,
+  Integer[0, 5000] $wcache_page_size = $::katello::qpid_wcache_page_size,
+  String $interface = $::katello::qpid_interface,
 ){
-  if $katello_user == undef {
-    fail('katello_user not defined')
-  } else {
-    Group['qpidd'] ->
-    User<|title == $katello_user|>{groups +> 'qpidd'}
+  include ::certs::qpid
+
+  class { '::qpid':
+    ssl                    => true,
+    ssl_cert_db            => $::certs::nss_db_dir,
+    ssl_cert_password_file => $::certs::qpid::nss_db_password_file,
+    ssl_cert_name          => 'broker',
+    interface              => $interface,
+    session_unacked        => $session_unacked,
+    wcache_page_size       => $wcache_page_size,
+    subscribe              => Class['certs::qpid'],
   }
 
+  # TODO: Is this still needed with proper containment?
+  Class['certs::qpid'] ~> Service['qpidd']
+
+  Group['qpidd'] ->
+  User<|title == $katello_user|>{groups +> 'qpidd'}
+
   qpid::config_cmd {'delete katello entitlements queue if bound to *.*':
     command  => "del queue ${candlepin_event_queue} --force",
     onlyif   => "list binding | grep ${candlepin_event_queue} | grep '*.*'",
-    ssl_cert => $client_cert,
-    ssl_key  => $client_key,
+    ssl_cert => $::certs::qpid::client_cert,
+    ssl_key  => $::certs::qpid::client_key,
   } ->
   qpid::config_cmd { 'create katello entitlements queue':
     command  => "add queue ${candlepin_event_queue} --durable",
     unless   => "queues ${candlepin_event_queue}",
-    ssl_cert => $client_cert,
-    ssl_key  => $client_key,
+    ssl_cert => $::certs::qpid::client_cert,
+    ssl_key  => $::certs::qpid::client_key,
   } ->
   qpid::config::bind { ['entitlement.created', 'entitlement.deleted', 'pool.created', 'pool.deleted', 'compliance.created']:
     queue    => $candlepin_event_queue,
     exchange => $candlepin_qpid_exchange,
```

Comment:
https://github.com/Katello/puppet-qpid/pull/71 ensures that the chaining is there automatically. When that's in we can also change the above ```config_cmd``` to a queue type.